Mathlib/Analysis/Distribution/TestFunction.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2025 Luigi Massacci. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Luigi Massacci
+-/
+
+import Mathlib.Analysis.Calculus.ContDiff.Defs
+import Mathlib.Topology.ContinuousMap.Bounded.Normed
+
+/-!
+# Continuously differentiable functions with compact support
+
+This file develops the basic theory of bundled `n`-times continuously differentiable functions
+with compact support contained in some open set `Ω`. More explicitly, given normed spaces `E`
+and `F`, an open set `Ω : Opens E` and `n : ℕ∞`, we are interested is the space `𝓓^{n}(Ω, F)` of
+maps `f : E → F` such that:
+
+- `f` is `n`-times continuously differentiable: `ContDiff ℝ n f`.
+- `f` has compact support: `HasCompactSupport f`.
+- the support of `f` is inside the open set `Ω`: `tsupport f ⊆ Ω`.
+
+This exists as a bundled type to equip it with the canonical LF topology induced by the inclusions
+`𝓓_{K}^{n}(Ω, F) → 𝓓^{n}(Ω, F)` (see `ContDiffMapSupportedIn`). The dual space is then the space of
+distributions, or "weak solutions" to PDEs, on `Ω`.
+
+## Main definitions
+
+- `TestFunction Ω F n`: the type of bundled `n`-times continuously differentiable
+  functions `E → F` with compact support contained in `Ω`.
+
+## Notation
+
+- `𝓓^{n}(Ω, F)`: the space of bundled `n`-times continuously differentiable functions `E → F`
+  with compact support contained in `Ω`.
+- `𝓓(Ω, F)`: the space of bundled smooth (infinitely differentiable) functions `E → F`
+  with compact support contained in `Ω`, i.e. `𝓓^{⊤}(Ω, F)`.
+
+## Tags
+
+distributions
+-/
+
+open TopologicalSpace SeminormFamily Set Function Seminorm UniformSpace
+open scoped BoundedContinuousFunction Topology NNReal
+
+variable (𝕜 : Type*) [NontriviallyNormedField 𝕜]
+variable {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E] (Ω : Opens E)
+variable (F : Type*) [NormedAddCommGroup F] [NormedSpace ℝ F]
+variable [NormedSpace 𝕜 F] [SMulCommClass ℝ 𝕜 F]
+variable {n : ℕ∞}
```

Comment:
If I see correctly, you're only using k, Omega and F being explicit for the next structure (but not any of the subsequent instances). Can you make them implicit here?
```suggestion
variable {𝕜 : Type*} [NontriviallyNormedField 𝕜]
  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E] {Ω : Opens E}
  {F : Type*} [NormedAddCommGroup F] [NormedSpace ℝ F] [NormedSpace 𝕜 F] [SMulCommClass ℝ 𝕜 F]
  {n : ℕ∞}
```